blueocean-dashboard/src/main/js/util/UrlUtils.js

```diff
@@ -173,11 +173,12 @@ export function getRestUrl({ organization, pipeline, branch, runId }) {
     const jenkinsUrl = require('../config').getJenkinsRootURL();
     let url = `${jenkinsUrl}/blue/rest/organizations/${encodeURIComponent(organizationName)}`;
     if (pipelineName) {
-        // pipelineName might include a folder path, don't escape it
+        // pipelineName might include a folder path, don't encode it
         url += `/pipelines/${pipelineName}`;
     }
     if (branch) {
-        url += `/branches/${encodeURIComponent(branch)}`;
+        // JENKINS-37712 branch needs to be double-encoded for some reason
+        url += `/branches/${encodeURIComponent(encodeURIComponent(branch))}`;
```

Comment:
@kzantow More of jenkins core question but I see why its this way. 

Basically, '/' in particular is used by Stapler to dispatch/route. So if you have it singly encoded then Stapler will see /feature/ux-1 as two separate tokens, feature and ux-1 so might try getFeature("ux-1"). For this reason, any url unfriendly names are encoded and stored in model. When such encoded names are used in URL, they are doubly encoded so that 'feature/ux-1' is treaded as one token with single encoding. Branch names are such example, where a branch name could have any url unfriendly characters and are encoded in core and kept as such.